src/objects.rs

```diff
@@ -691,6 +692,48 @@ impl GetObjectType for VehicleJourney {
     }
 }
 
+#[derive(Debug, Error)]
+pub enum StopTimeError {
+    #[error("duplicate stop_sequence '{duplicated_sequence}' for the trip '{vj_id}'")]
+    DuplicateStopSequence {
+        vj_id: String,
+        duplicated_sequence: u32,
+    },
+    #[error("incoherent stop times '{first_incorrect_sequence}' at time '{first_incorrect_time}' for the trip '{vj_id}'")]
+    IncoherentStopTimes {
+        vj_id: String,
+        first_incorrect_sequence: u32,
+        first_incorrect_time: Time,
+    },
+}
+
+impl VehicleJourney {
+    #[allow(clippy::suspicious_operation_groupings)]
```

Comment:
You can put this on line 723 to target more precisely the impacted piece of code.